network/f5/bigip_virtual_server.py

```diff
@@ -0,0 +1,438 @@
+            if not vs_exists(api, name):
+                if (not destination) or (not port):
+                    module.fail_json(msg="both destination and port must be supplied to create a VS")
+                if not module.check_mode:
+                    # a bit of a hack to handle concurrent runs of this module.
+                    # even though we've checked the pool doesn't exist,
+                    # it may exist by the time we run create_pool().
+                    # this catches the exception and does something smart
+                    # about it!
+                    try:
+                        vs_create(api,name,destination,port,pool)
+                        set_profiles(api,name,all_profiles)
+                        set_snat(api,name,snat)
+                        set_description(api,name,description)
+                        set_default_persistence_profiles(api,name,default_persistence_profile)
+                        result = {'changed': True}
+                    except bigsuds.OperationFailed, e:
+                            raise Exception('Error on creating Virtual Server : %s' % e)
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+            else:
+                update = True
+            if update:
+                # VS exists
+                if not module.check_mode:
+                    # Have a transaction for all the changes
+                    try:
+                        api.System.Session.start_transaction()
+                        result['changed']|=set_destination(api,name,fq_name(partition,destination),port)
+                        result['changed']|=set_pool(api,name,pool)
+                        result['changed']|=set_description(api,name,description)
+                        result['changed']|=set_snat(api,name,snat)
+                        result['changed']|=set_profiles(api,name,all_profiles)
+                        result['changed']|=set_default_persistence_profiles(api,name,default_persistence_profile)
+                        api.System.Session.submit_transaction()
+                    except Exception,e:
+                        raise Exception("Error on updating Virtual Server : %s" % e)
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+
+        elif state in ('disabled', 'enabled'):
+            if name is None:
+                module.fail_json(msg="name parameter required when " \
+                                     "state=enabled/disabled")
+            if not module.check_mode:
+                pass
+            else:
+                # check-mode return value
+                result = {'changed': True}
+
+    except Exception, e:
+        module.fail_json(msg="received exception: %s" % e)
+
+    module.exit_json(**result)
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.f5 import *
+main()
```

Comment:
Please include the module execution in a conditional per the checklist http://docs.ansible.com/ansible/developing_modules.html#module-checklist

```
if __name__ == '__main__':
    main()
```